File: Courzelo-backend-LMS/Courzelo/src/main/java/org/example/courzelo/dto/responses/ClassRoomResponse.java
package org.example.courzelo.dto.responses;

import lombok.Builder;
import lombok.Data;
import org.example.courzelo.dto.QuizDTO;

import java.util.List;

@Data
@Builder
public class ClassRoomResponse {
    private String id;
    private String name;
    private String description;
    private String course;
    private int credit;
    private String teacher;
    private List<String> students;
    private String group;
    private String institutionID;
    private List<ClassRoomPostResponse> posts;
    private List<QuizDTO> quizzes;
}

File: courzelo-dashboard-LMS-main/src/app/shared/models/institution/ClassRoomResponse.ts
import {ClasssRoomPostResponse} from './ClasssRoomPostResponse';
import {Quiz} from '../Quiz';
import {SafeUrl} from '@angular/platform-browser';

export interface ClassRoomResponse {
    imageSrc: SafeUrl;
    id?: string;
    name?: string;
    description?: string;
    course?: string;
    credit?: number;
    teacher?: string;
    group?: string;
    institutionID?: string;
    created?: Date;
    posts?: ClasssRoomPostResponse[];
    quizzes?: Quiz[];
}
